packages/protocol/test-sol/unit/common/EpochManager.t.sol

```diff
@@ -469,3 +468,60 @@ contract EpochManagerTest_sendValidatorPayment is EpochManagerTest {
     assertEq(epochManagerBalanceAfter, epochManagerBalanceBefore - paymentAmount);
   }
 }
+
+contract EpochManagerTest_getEpochOfBlock is EpochManagerTest {
+  function test_shouldReturnTheEpochInfoOfSpecifiedEpoch() public {
+    initializeEpochManagerSystem();
+    uint256 _startingEpochNumber = epochManager.getCurrentEpochNumber();
+
+    (
+      uint256 _initialFirstBlock,
+      uint256 _initialLastBlock,
+      uint256 _initialStartTimestamp,
+      uint256 _initialRewardBlock
+    ) = epochManager.getCurrentEpoch();
+
+    (
+      uint256 _firstBlock,
+      uint256 _lastBlock,
+      uint256 _startTimestamp,
+      uint256 _rewardBlock
+    ) = epochManager.getEpochInfoOfEpoch(_startingEpochNumber);
+
+    assertEq(_initialFirstBlock, _firstBlock);
+    assertEq(_initialLastBlock, _lastBlock);
+    assertEq(_initialStartTimestamp, _startTimestamp);
+    assertEq(_initialRewardBlock, _rewardBlock);
+
+    travelEpochL2(vm);
```

Comment:
Maybe split this into separate test cases, each ending in a series of asserts?

Also, any other test cases to add? E.g., after N epochs, asking for `current - N` still returns the old data, `current` returns the new data.